### ISSUE TITLE
2.0.0rc2 POST /ca_certificates returns a 500

### ISSUE THREAD

**Issue by hbagdi** [Kong/kong#5446]
### Summary

POST /ca_certificates succeeds in Kong 1.4 but fails in 2.0.0rc2

### Steps To Reproduce

1. Install Kong 1.4
2. Run `http :8001/ca_certificates cert=@ca_cert.crt`, it will succeed 
3. Install 2.0.0rc2
4. Run `http :8001/ca_certificates cert=@ca_cert.crt`, it fails.

### Additional Details & Logs

- Kong version (`$ kong version`)
- Kong debug-level startup logs (`$ kong start --vv`)
- Kong error logs (`<KONG_PREFIX>/logs/error.log`)

```
2020/01/15 05:35:07 [error] 29#0: *6090 [lua] api_helpers.lua:422: handle_error(): /usr/local/share/lua/5.1/lapis/application.lua:401: /usr/local/share/lua/5.1/resty/openssl/asn1.lua:14: attempt to perform arithmetic on a nil value
stack traceback:
	/usr/local/share/lua/5.1/resty/openssl/asn1.lua: in function 'yday'
	/usr/local/share/lua/5.1/resty/openssl/asn1.lua:43: in function 'to'
	/usr/local/share/lua/5.1/resty/openssl/x509/init.lua:284: in function 'get_not_after'
	...hare/lua/5.1/kong/db/schema/entities/ca_certificates.lua:29: in function 'fn'
	/usr/local/share/lua/5.1/kong/db/schema/init.lua:1186: in function 'run_entity_check'
	/usr/local/share/lua/5.1/kong/db/schema/init.lua:1262: in function 'run_checks'
	/usr/local/share/lua/5.1/kong/db/schema/init.lua:1278: in function 'run_entity_checks'
	/usr/local/share/lua/5.1/kong/db/schema/init.lua:1817: in function 'validate_insert'
	/usr/local/share/lua/5.1/kong/db/dao/init.lua:334: in function 'check_insert'
	/usr/local/share/lua/5.1/kong/db/dao/init.lua:946: in function 'insert_entity'
	/usr/local/share/lua/5.1/kong/api/endpoints.lua:375: in function 'fn'
	/usr/local/share/lua/5.1/kong/api/api_helpers.lua:257: in function </usr/local/share/lua/5.1/kong/api/api_helpers.lua:240>

stack traceback:
	[C]: in function 'error'
	/usr/local/share/lua/5.1/lapis/application.lua:401: in function 'handler'
	/usr/local/share/lua/5.1/lapis/application.lua:130: in function 'resolve'
	/usr/local/share/lua/5.1/lapis/application.lua:167: in function </usr/local/share/lua/5.1/lapis/application.lua:165>
	[C]: in function 'xpcall'
	/usr/local/share/lua/5.1/lapis/application.lua:173: in function 'dispatch'
	/usr/local/share/lua/5.1/lapis/nginx.lua:230: in function 'serve'
	/usr/local/share/lua/5.1/kong/init.lua:1099: in function 'admin_content'
	content_by_lua(nginx-kong.conf:234):2: in main chunk, client: 172.17.0.1, server: kong_admin, request: "POST /ca_certificates HTTP/1.1", host: "localhost:8001"
172.17.0.1 - - [15/Jan/2020:05:35:07 +0000] "POST /ca_certificates HTTP/1.1" 500 42 "-" "HTTPie/2.0.0"
```

- Kong configuration (the output of a GET request to Kong's Admin port - see
  https://docs.konghq.com/latest/admin-api/#retrieve-node-information)
- Operating system: Docker

ca_cert.crt:

```
-----BEGIN CERTIFICATE-----
MIIEvjCCAqagAwIBAgIJALabx/Nup200MA0GCSqGSIb3DQEBCwUAMBMxETAPBgNV
BAMMCFlvbG80Mi4xMCAXDTE5MDkxNTE2Mjc1M1oYDzIxMTkwODIyMTYyNzUzWjAT
MREwDwYDVQQDDAhZb2xvNDIuMTCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoC
ggIBANIW67Ay0AtTeBY2mORaGet/VPL5jnBRz0zkZ4Jt7fEq3lbxYaJBnFI8wtz3
bHLtLsxkvOFujEMY7HVd+iTqbJ7hLBtK0AdgXDjf+HMmoWM7x0PkZO+3XSqyRBbI
YNoEaQvYBNIXrKKJbXIU6higQaXYszeN8r3+RIbcTIlZxy28msivEGfGTrNujQFc
r/eyf+TLHbRqh0yg4Dy/U/T6fqamGhFrjupRmOMugwF/BHMH2JHhBYkkzuZLgV2u
7Yh1S5FRlh11am5vWuRSbarnx72hkJ99rUb6szOWnJKKew8RSn3CyhXbS5cb0QRc
ugRc33p/fMucJ4mtCJ2Om1QQe83G1iV2IBn6XJuCvYlyWH8XU0gkRxWD7ZQsl0bB
8AFTkVsdzb94OM8Y6tWI5ybS8rwl8b3r3fjyToIWrwK4WDJQuIUx4nUHObDyw+KK
+MmqwpAXQWbNeuAc27FjuJm90yr/163aGuInNY5Wiz6CM8WhFNAi/nkEY2vcxKKx
irSdSTkbnrmLFAYrThaq0BWTbW2mwkOatzv4R2kZzBUOiSjRLPnbyiPhI8dHLeGs
wMxiTXwyPi8iQvaIGyN4DPaSEiZ1GbexyYFdP7sJJD8tG8iccbtJYquq3cDaPTf+
qv5M6R/JuMqtUDheLSpBNK+8vIe5e3MtGFyrKqFXdynJtfHVAgMBAAGjEzARMA8G
A1UdEwQIMAYBAf8CAQAwDQYJKoZIhvcNAQELBQADggIBAK0BmL5B1fPSMbFy8Hbc
/ESEunt4HGaRWmZZSa/aOtTjhKyDXLLJZz3C4McugfOf9BvvmAOZU4uYjfHTnNH2
Z3neBkdTpQuJDvrBPNoCtJns01X/nuqFaTK/Tt9ZjAcVeQmp51RwhyiD7nqOJ/7E
Hp2rC6gH2ABXeexws4BDoZPoJktS8fzGWdFBCHzf4mCJcb4XkI+7GTYpglR818L3
dMNJwXeuUsmxxKScBVH6rgbgcEC/6YwepLMTHB9VcH3X5VCfkDIyPYLWmvE0gKV7
6OU91E2Rs8PzbJ3EuyQpJLxFUQp8ohv5zaNBlnMb76UJOPR6hXfst5V+e7l5Dgwv
Dh4CeO46exmkEsB+6R3pQR8uOFtubH2snA0S3JA1ji6baP5Y9Wh9bJ5McQUgbAPE
sCRBFoDLXOj3EgzibohC5WrxN3KIMxlQnxPl3VdQvp4gF899mn0Z9V5dAsGPbxRd
quE+DwfXkm0Sa6Ylwqrzu2OvSVgbMliF3UnWbNsDD5KcHGIaFxVC1qkwK4cT3pyS
58i/HAB2+P+O+MltQUDiuw0OSUFDC0IIjkDfxLVffbF+27ef9C5NG81QlwTz7TuN
zeigcsBKooMJTszxCl6dtxSyWTj7hJWXhy9pXsm1C1QulG6uT4RwCa3m0QZoO7G+
6Wu6lP/kodPuoNubstIuPdi2
-----END CERTIFICATE-----
```

**Comment by bungle** [comment]
Ping @fffonion

**Comment by bungle** [comment]
Off topic:
I would URL of `ca_certificates`  being `ca-certificates` instead.

**Comment by fffonion** [comment]
Thanks for reporting this, it's indeed a bug in asn1 time parsing when year is YYYY instead of YY.

**Comment by hbagdi** [comment]
I can reproduce this issue on 2.0.0.

**Comment by fffonion** [comment]
Looks like #5448 is not in master or 2.0.0

**Comment by hishamhm** [comment]
@fffonion this was merged into `next` which already had commits made for 2.1 when 2.0.0rc2 was released, so `next` wasn't pulled into 2.0.0 final. We should have merged the fix straight into `release/2.0.0`. I did have a milestone tag for 2.0.0 on #5448 but it was already merged so I missed that it was pending for the actual release, my bad. Now we need to cherry-pick this bump into `master`.

(Sometimes I wish we used development branches named after `x.y` release series instead of `next`, which always gets confusing around feature freezes!)
